### PR TITLE
Add support for playing a track list only once

### LINKED ISSUE #2698: Extinction theme is played on loop
Found that on game lose the extinction theme is still actually looping despite its loop property is disabled in the import setting. Underlying cause most likely is in the Jukebox class itself. Need a way to play a list of track without repeating.

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

Track list can now be non-repeating. Not really confident with the implementation so feel free to suggest a better approach.

**Related Issues**

Closes #2698 

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/general/Jukebox.cs [hunk 1]
```diff
@@ -538,6 +543,8 @@ private void OnCategoryEnded()
             {
                 foreach (var track in list.Tracks)
                 {
+                    track.PlayedOnce = false;
```
Comment (hhyyrylainen) [inline src/general/Jukebox.cs:546]: I found a bug regarding this (this reset needs to *always* happen when `category` is not null.
If you go into a new game and go extinct, the track plays fine once, but if you exit to menu and then start a new game where you go extinct again, you don't hear the extinct track again even though you should.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Still does not work:
```
Jukebox now playing from: Extinction
Jukebox: starting track: res://assets/sounds/extinct2.ogg position: 63.46594
```
Did it work for you?

**Comment by Kasterisk** [comment]
Oh do you mean the track position should reset instead of continuing? From your jukebox log it seems the track is resumed at nearing the end which has quite a long silence in the audio stream but not yet ended, so that may give the impression the track never plays once it does.

**Comment by FuJa0815** [comment]
Couldn't you add a property like `NeedsToBeginFromStart`?

**Comment by Kasterisk** [comment]
The actual solution would be adding a new member in MusicCategory.ReturnType enum to not continue.

**Comment by hhyyrylainen** [comment]
I guess I didn't implement that? I guess there wasn't a music category before that didn't continue? Isn't the main menu category like that, or does that continue as well?

**Comment by Kasterisk** [comment]
Nope, there is only one return type that is to continue. All categories currently continues including the main menu.

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**